packages/hardhat-core/src/internal/hardhat-network/provider/modules/debug.ts

```diff
@@ -32,7 +37,31 @@ export class DebugModule {
   private _traceTransactionParams(
     params: any[]
   ): [Buffer, RpcDebugTracingConfig] {
-    return validateParams(params, rpcHash, rpcDebugTracingConfig);
+    const validatedParams = validateParams(
+      params,
+      rpcHash,
+      rpcDebugTracingConfig
+    );
+
+    this._throwErrIfTracerIsNotSupported(validatedParams[1]);
+
+    return validatedParams;
+  }
+
+  private _throwErrIfTracerIsNotSupported(config: RpcDebugTracingConfig) {
```

Comment:
Nitpicking, but I would name the function according to its purpose, maybe something like `_validateTracer`.